### PR TITLE
reset height of children

### PR DESCRIPTION
**Pull request by willmcgugan**
Reset height of children in render.

Fix for https://github.com/Textualize/rich/issues/2042

When rendering an object with a height, that height is passed to the renderable via options. The problem was that any renderables yielded by `__rich_console__` were passed that same height, which was not intended.

This change ensures that height is reset for these sub objects.

Also removes tabulate.py which was marked for deprecation a while ago.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,14 +24,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fix syntax lexer guessing.
 - Fixed Pretty measure not respecting expand_all https://github.com/Textualize/rich/issues/1998
 - Collapsed definitions for single-character spinners, to save memory and reduce import time.
-- Fix print_json indent type in __init__.py
+- Fix print_json indent type in **init**.py
```
Comment (darrenburns) [inline CHANGELOG.md:27]: Not exactly an issue but just curious - is this some markdown formatter thing that caused this?

File: rich/console.py [hunk 2]
```diff
@@ -224,6 +224,16 @@ def update_height(self, height: int) -> "ConsoleOptions":
         options.max_height = options.height = height
         return options
 
+    def reset_height(self) -> "ConsoleOptions":
+        """Resets the height to None.
```
Comment (darrenburns) [inline rich/console.py:228]: The docstring makes it sound like the method is doing self.height = None
```suggestion
        """Returns a copy of the ConsoleOptions with height=None
```

**Review by darrenburns — APPROVED** [review]
Minor docstring change suggested, take it or leave it :)

File: CHANGELOG.md (hunk 1 above)
Comment (willmcgugan) [reply CHANGELOG.md:27]: Yeah, I think that may have been vscode. I need to put it inside backticks.

File: rich/console.py (hunk 2 above)
Comment (willmcgugan) [reply rich/console.py:228]: Good point.

I had auto-merge on. Feel free to "request changes" if there is a worthwhile change.

**Pull request merged**